src/test/java/org/assertj/core/api/future/CompletableFutureAssert_hasNotFailed_Test.java

```diff
@@ -55,6 +56,7 @@ public void should_fail_if_completable_future_has_failed() {
     future.completeExceptionally(new RuntimeException());
 
     assertThatThrownBy(() -> assertThat(future).hasNotFailed()).isInstanceOf(AssertionError.class)
-                                                               .hasMessage(shouldNotHaveFailed(future).create());
+                                                               .hasMessageMatching(shouldNotHaveFailed(future).create(emptyDescription(),
+                                                                                                              new FailedCompletableFutureRepresentation()));
```

Comment:
I'm not too keen about having a `FailedCompletableFutureRepresentation` that kind of duplicate the production code.

I believe we should simply verify that some subpart of the stack trace is present  (a subpart that can only be in the stack trace) either with `hasMessageMatching` or `hasMessageContaining`, the rationale is that if such subpart exists then the whole stack trace should be there too.